### PR TITLE
Core distributions from dust

### PR DESCRIPTION
**Pull request by richfitz**
This PR pulls in code from dust with minimal changes, and wraps up a basic RNG around `Math.random()` - we'll probably want to bring our own better generator at some point but this should be ok for now.

The distributions here should allow implementation of all the short course models

There's a bit of prior work in this space, but none that seemed (eventually) seedable, supporting good random-access to distributions, or with types ready to go. The closest is random, which we could contribute to, but see for example the implementation of binomial (which is one of the ones we depend most strongly on in practice) https://github.com/transitive-bullshit/random/blob/master/src/distributions/binomial.ts

Eventually it would be good to copy some of this over to use wasm of some flavour, but that's a problem for later.

Final webpack bundle is only 3.4k so that feels like it'd be ok to pull in to wodin :)

### CODE REVIEW

File: src/binomial.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    }
+    const q = p <= 0.5 ? p : 1 - p;
+    const draw = n * q >= 10 ?
+        btrs(state, n, q) :
+        inversion(state, n, q);
+    return p <= 0.5 ? draw : n - draw;
+}
+
+function btrs(state: RngState, n: number, p: number) {
+    // This is spq in the paper.
```
Comment (hillalex) [inline src/binomial.ts:30]: which paper?

File: test/binomial.test.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import {binomial} from "../src/binomial";
+import {RngStateBuiltin} from "../src/state-builtin";
+
+import {approxEqual, mean, repeat, variance} from "./helpers";
+import "./matchers";
+
+describe("binomial random numbers", () => {
+    const state = new RngStateBuiltin();
+    it("generates with expected mean and variance", () => {
```
Comment (hillalex) [inline test/binomial.test.ts]:
```suggestion
    it("generates with expected mean and variance when n * p is large", () => {
```

File: test/binomial.test.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+        expect(repeat<number>(() => binomial(state, 0, p), m))
+            .toEqual(Array(m).fill(0));
+        expect(repeat<number>(() => binomial(state, n, 0), m))
+            .toEqual(Array(m).fill(0));
+        expect(repeat<number>(() => binomial(state, n, 1), m))
+            .toEqual(Array(m).fill(n));
+    });
+
+    it("generates binomial numbers via inversion", () => {
```
Comment (hillalex) [inline test/binomial.test.ts]:
```suggestion
    it("generates with expected mean and variance when n * p is small", () => {
```

File: test/exponential.test.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+describe("exponential random numbers", () => {
+    const state = new RngStateBuiltin();
+    it("generates std distribution with expected mean and variance", () => {
+        const n = 1000000;
+        const res = repeat<number>(() => randomExponential(state), n);
+        expect(mean(res)).toApproxEqual(1, 1e-2);
+        expect(variance(res)).toApproxEqual(1, 1e-2);
+    });
+
+    it("generates std distribution with expected mean and variance", () => {
```
Comment (hillalex) [inline test/exponential.test.ts]:
```suggestion
    it("generates general distribution with expected mean and variance", () => {
```

File: .github/workflows/docs.yml [hunk 5]
```diff
@@ -2,7 +2,7 @@ name: docs
 
 on:
   push:
-    branches: [main, master]
+    branches: [main, master, mrc-3365]
```
Comment (hillalex) [inline .github/workflows/docs.yml]: I guess this was for testing and can be removed

File: src/binomial.ts (hunk 1 above)
Comment (hillalex) [inline src/binomial.ts:33]: which paper? would be good to have a link to the source for these algorithms!

Comment (richfitz) [reply src/binomial.ts:33]: hah, yeah, will copy that over :facepalm:

**Review by hillalex — APPROVED** [review]

**Pull request merged**